var.mk

```diff
@@ -1,5 +1,27 @@
 # to be included by Makefile
 
```

Comment:
Because of readability/flexibility I prefer:
```
STACK ?= $(shell command -v stack > /dev/null)
STACK_EXEC :=
STACK_TARGET :=
STACK_UPGRADE_TARGET :=
STACK_DEPENDENCIES_FLAGS :=
ifneq ($(STACK),)
    STACK_EXEC := $(STACK) exec --
    ...
endif
```

Last but not least `printf` should be used instead of `echo` - there are many different echo implementations/shell builtins and it emits per default a redundant linefeed at the EOL, which may cause add. headaches from time to time ...